iceoryx_posh/include/iceoryx_posh/popo/modern_api/base_subscriber.hpp

```diff
@@ -31,12 +31,16 @@ using uid_t = UniquePortId;
 template <typename T, typename port_t = iox::SubscriberPortUserType>
 class BaseSubscriber : public Condition
 {
-  public:
+  protected:
+    BaseSubscriber(const capro::ServiceDescription& service);
     BaseSubscriber(const BaseSubscriber& other) = delete;
     BaseSubscriber& operator=(const BaseSubscriber&) = delete;
     BaseSubscriber(BaseSubscriber&& rhs) = delete;
     BaseSubscriber& operator=(BaseSubscriber&& rhs) = delete;
-    ~BaseSubscriber();
+    virtual ~BaseSubscriber();
+
+    // Required for testing. Allows BaseSubscriber instance to be constructed without a PoshRuntime.
+    BaseSubscriber() noexcept {};
```

Comment:
@ithier I think you should use `= default` here instead of providing an empty ctor here?
`BaseSubscriber() = default`.

Reply:
Yeah, that's what I did initially but the compiler was complaining about the function being deleted when I did this for some reason and I couldn't immediately see why. I'll try fix it.

Reply:
If a member variable has no default constructor then defaulting a constructor leads to a deleted constructor

Reply:
@ithier the problem seems to be the `SubscriberSampleDeleter` which has no default constructor. As soon as you create a ctor with an argument the default ctor is deleted. Maybe adding there a default ctor like `SubscriberSampleDeleter() = default;` solves the issue

Reply:
`SubscriberPortUser` has no default constructor either.
Is there a reason to prefer using `'=default` in cases like this ?

Reply:
If you need a test coverage of 100 % then you have to write an additional test when you implement it but if the compiler generates that code and is already certified you have to do nothing.

Reply:
Unless I misunderstood, I guess we can't take advantage of that here because of the `SubscriberPortUser` member.